context/app/static/js/components/detailPage/visualization/VisualizationNotebookButton/VisualizationNotebookButton.tsx

```diff
@@ -1,43 +1,69 @@
-import React from 'react';
-import GetAppRoundedIcon from '@mui/icons-material/GetAppRounded';
-import postAndDownloadFile from 'js/helpers/postAndDownloadFile';
+import React, { useMemo } from 'react';
+
+import Download from '@mui/icons-material/Download';
 
-import { WhiteBackgroundIconButton } from 'js/shared-styles/buttons';
+import withDropdownMenuProvider from 'js/shared-styles/dropdowns/DropdownMenuProvider/withDropdownMenuProvider';
 import { useTrackEntityPageEvent } from 'js/components/detailPage/useTrackEntityPageEvent';
 import { useSnackbarActions } from 'js/shared-styles/snackbars';
-import { StyledSecondaryBackgroundTooltip } from './style';
+import { WorkspacesIcon } from 'js/shared-styles/icons';
+import postAndDownloadFile from 'js/helpers/postAndDownloadFile';
+import NewWorkspaceDialog from 'js/components/workspaces/NewWorkspaceDialog';
+import { useCreateWorkspaceForm } from 'js/components/workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import IconDropdownMenu from 'js/shared-styles/dropdowns/IconDropdownMenu';
 
-const title = 'Download Jupyter Notebook';
+const tooltip = 'Launch new workspace or download a Jupyter notebook for this visualization.';
 
 interface VisualizationNotebookButtonProps {
   uuid: string;
+  hubmap_id: string;
+  mapped_data_access_level: string;
 }
 
-function VisualizationNotebookButton({ uuid }: VisualizationNotebookButtonProps) {
+function VisualizationNotebookButton({ uuid, hubmap_id, mapped_data_access_level }: VisualizationNotebookButtonProps) {
   const trackEntityPageEvent = useTrackEntityPageEvent();
   const { toastError } = useSnackbarActions();
 
+  const { setDialogIsOpen, ...rest } = useCreateWorkspaceForm({
+    defaultName: hubmap_id,
+    defaultTemplate: 'visualization',
+  });
+
+  const downloadNotebook = useMemo(() => {
+    return () => {
+      trackEntityPageEvent({ action: `Vitessce / ${tooltip}` });
+      postAndDownloadFile({
+        url: `/notebooks/entities/dataset/${uuid}.ws.ipynb`,
+        body: {},
+      })
+        .then(() => {
+          // Do nothing
+        })
+        .catch(() => {
+          toastError('Failed to download Jupyter Notebook');
+        });
+    };
+  }, [uuid, toastError, trackEntityPageEvent]);
+
+  const menuOptions = [
```

Comment:
@NickAkhmetov what do you think of this pattern?

Reply:
This approach should work, but presents some minor maintainability concerns. 

An alternative pattern would be to provide these options as children of the IconDropdownMenu instead of passing in this array, which might keep usage a bit more open-ended (i.e. if we want to do a one-off style for some option in the future, it'd be easier to apply those styles with that pattern than it would be to extend the `IconDropdownMenuOption` component down the line)

Reply:
I like that idea as well.